sui_programmability/examples/games/sources/drand_based_scratch_card.move

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A basic game that depends on randomness from drand (chained mode).
+///
+/// Anyone can create a new game by depositing X*100 SUIs as a reward, and setting the current drand round as the base
+/// round. This creates two objects:
+/// - Game - an immutable object that includes all parameters to be used when buying tickets.
+/// - Reward - a shared object that holds the reward. It can be withdrawn by any winner ("first come, first served").
```

Comment:
I'm happy to publish this game type, but ideally due to this FCFS rule we shouldn't recommend this pattern in practice for most real world applications. We need a warning message at the top of this file, if we keep this game.

Ideally, for the future we need a multi-player game with sortition and challenge period, to ensure one winner.